source/includes/fundamentals/code-snippets/change-streams/ConfiguredChangeStream.java

```diff
@@ -0,0 +1,47 @@
+package org.example;
+
+import com.mongodb.client.ChangeStreamIterable;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.changestream.FullDocument;
+import com.mongodb.client.model.changestream.FullDocumentBeforeChange;
+import org.bson.Document;
+
+public class ConfiguredChangeStream extends Thread {
+    private MongoCollection<Document> collection;
+    private final FullDocument fullDocument;
+    private final FullDocumentBeforeChange fullDocumentBeforeChange;
+    private String name;
+
+    public ConfiguredChangeStream(MongoCollection<Document> collection,
+                                  FullDocument fullDocument,
+                                  FullDocumentBeforeChange fullDocumentBeforeChange) {
+        this.collection = collection;
+        this.fullDocument = fullDocument;
+        this.fullDocumentBeforeChange = fullDocumentBeforeChange;
+        this.name = fullDocument != null ? fullDocument.name() : fullDocumentBeforeChange.name();
+    }
+    public void run() {
+        ChangeStreamIterable<Document> changeStream = this.collection.watch();
+//                .fullDocumentBeforeChange(this.fullDocumentBeforeChange);
```

Comment:
Is this commented code left in intentionally?

Reply:
Nope, good catch!